src/agent/src/rpc.rs

```diff
@@ -1432,23 +1472,39 @@ fn update_container_namespaces(
     // Else set this to empty string so that a new pid namespace is
     // created for the container.
     if sandbox_pidns && sandbox.sandbox_pidns.is_some() {
-        pid_ns.path = String::from(sandbox.sandbox_pidns.as_ref().unwrap().path.as_str());
+        pid_ns.path = String::from(
+            sandbox
+                .sandbox_pidns
+                .as_ref()
+                .ok_or("failed to get sandbox ref")
+                .map_err(|e| anyhow!(e))?
+                .path
+                .as_str(),
+        );
```

Comment:
You have an `is_some()` check above, so again you want `if let` rather than returning an error.